corehq/apps/userreports/views.py

```diff
@@ -129,23 +132,105 @@ def decorated(request, domain, *args, **kwargs):
+class BaseUserConfigReportsView(BaseDomainView):
+    section_name = ugettext_lazy("Configurable Reports")
 
+    @property
+    def main_context(self):
+        static_reports = list(StaticReportConfiguration.by_domain(self.domain))
+        static_data_sources = list(StaticDataSourceConfiguration.by_domain(self.domain))
+        context = super(BaseUserConfigReportsView, self).main_context
+        context.update({
+            'reports': ReportConfiguration.by_domain(self.domain) + static_reports,
+            'data_sources': DataSourceConfiguration.by_domain(self.domain) + static_data_sources,
+        })
+        return context
 
-@login_and_domain_required
-@toggles.USER_CONFIGURABLE_REPORTS.required_decorator()
-def edit_report(request, domain, report_id):
-    config, is_static = get_report_config_or_404(report_id, domain)
-    return _edit_report_shared(request, domain, config, read_only=is_static)
+    @property
+    def section_url(self):
+        return reverse(UserConfigReportsHomeView.urlname, args=(self.domain,))
 
+    @property
+    def page_url(self):
+        return reverse(self.urlname, args=(self.domain,))
+
+    @use_bootstrap3
+    @method_decorator(toggles.USER_CONFIGURABLE_REPORTS.required_decorator())
+    def dispatch(self, request, *args, **kwargs):
+        return super(BaseUserConfigReportsView, self).dispatch(request, *args, **kwargs)
 
-@login_and_domain_required
-@toggles.USER_CONFIGURABLE_REPORTS.required_decorator()
-def create_report(request, domain):
-    return _edit_report_shared(request, domain, ReportConfiguration(domain=domain))
+
+class UserConfigReportsHomeView(BaseUserConfigReportsView):
+    urlname = 'configurable_reports_home'
+    template_name = 'userreports/configurable_reports_home.html'
+    page_title = ugettext_lazy("Reports Home")
+
+    @property
+    def page_context(self):
+        return {}
+
+
+class BaseEditConfigReportView(BaseUserConfigReportsView):
+    template_name = 'userreports/edit_report_config.html'
+
+    @property
+    def report_id(self):
+        return self.kwargs.get('report_id')
+
+    @property
+    def page_url(self):
+        if self.report_id:
+            return reverse(self.urlname, args=(self.domain, self.report_id,))
+        return super(BaseEditConfigReportView, self).page_url
+
+    @property
+    def page_context(self):
+        return {
+            'form': self.edit_form,
+            'report': self.config,
+        }
+
+    @property
+    @memoized
+    def config(self):
+        if self.report_id is None:
+            return ReportConfiguration(domain=self.domain)
+        try:
+            return get_report_configs([self.report_id], self.domain)[0]
+        except ReportConfigurationNotFoundError:
+            raise Http404()
+
+    @property
+    def read_only(self):
+        return _config_id_is_static(self.report_id) if self.report_id is not None else False
```

Comment:
should rename this function to not be "private" if you're going to import it from another module